packages/firestore/src/model/field_value.ts

```diff
@@ -313,6 +331,13 @@ export class StringValue extends FieldValue {
     }
     return this.defaultCompareTo(other);
   }
+
+  byteSize(): number {
+    // See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures:
+    // "JavaScript's String type is [...] a set of elements of 16-bit unsigned
+    // integer values"
+    return this.internalValue.length * 2;
```

Comment:
Note that if we're going for the on-disk size, this will probably (not 100% sure about IndexedDb) be UTF-8 encoded which means most characters will be 1-byte, not 2.